### PR TITLE
Implement ProxifyHostFile.evict()

### LINKED ISSUE #861: AttributeError: 'ProxifyHostFile' object has no attribute 'evict'
After changes introduced by https://github.com/dask/distributed/pull/5543, `dask_cuda/tests/test_proxify_host_file.py::test_worker_force_spill_to_disk` is now failing:

```python
___________________________________________ test_worker_force_spill_to_disk ____________________________________________

args = (), kwargs = {}, loop = <tornado.platform.asyncio.AsyncIOLoop object at 0x7f7314ba0580>
injected_func = functools.partial(<function test_worker_force_spill_to_disk at 0x7f7314b723a0>)
cor = functools.partial(<function test_worker_force_spill_to_disk at 0x7f7314b723a0>)

    def test_func(*args, **kwargs):
        with clean() as loop:
            injected_func = functools.partial(func, *args, **kwargs)
            if iscoroutinefunction(func):
                cor = injected_func
            else:
                cor = gen.coroutine(injected_func)

>           loop.run_sync(cor, timeout=timeout)

../distributed/distributed/utils_test.py:808:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../miniconda3/envs/rn-220217/lib/python3.8/site-packages/tornado/ioloop.py:530: in run_sync
    return future_cell[0].result()
dask_cuda/tests/test_proxify_host_file.py:404: in test_worker_force_spill_to_disk
    await client.submit(f)
../distributed/distributed/client.py:287: in _result
    raise exc.with_traceback(tb)
dask_cuda/tests/test_proxify_host_file.py:399: in f
    await w.memory_monitor()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   weight = self.data.evict()
E   AttributeError: 'ProxifyHostFile' object has no attribute 'evict'

../distributed/distributed/worker.py:3752: AttributeError
```

@shwina do you have the time and will to look at this? Otherwise I'll do so.

### PR DESCRIPTION
**Pull request by madsbk**
Closes #861

### CODE REVIEW

File: dask_cuda/proxify_host_file.py [hunk 1]
```diff
@@ -583,25 +583,44 @@ def oom(nbytes: int) -> bool:
                 mr = rmm.mr.FailureCallbackResourceAdaptor(current_mr, oom)
                 rmm.mr.set_current_device_resource(mr)
 
+    def evict(self) -> int:
+        """Manually evict 1% of host limit.
+
+        Dask use this to trigger CPU-to-Disk spilling. We don't know how much
```
Comment (pentschev) [inline dask_cuda/proxify_host_file.py:589]:
```suggestion
        Dask uses this to trigger CPU-to-Disk spilling. We don't know how much
```

File: dask_cuda/proxify_host_file.py [hunk 2]
```diff
@@ -583,25 +583,44 @@ def oom(nbytes: int) -> bool:
                 mr = rmm.mr.FailureCallbackResourceAdaptor(current_mr, oom)
                 rmm.mr.set_current_device_resource(mr)
 
+    def evict(self) -> int:
+        """Manually evict 1% of host limit.
```
Comment (pentschev) [inline dask_cuda/proxify_host_file.py:587]: I understand this was already like this before and not quite sure how useful that would be, but should the 1% value be configurable somehow?

File: dask_cuda/proxify_host_file.py [hunk 3]
```diff
@@ -583,25 +583,44 @@ def oom(nbytes: int) -> bool:
...
+        )
+        gc.collect()
+        return ret if ret > 0 else -1
+
     @property
     def fast(self):
-        """Dask use this to trigger CPU-to-Disk spilling"""
+        """Alternative access to `.evict()` used by Dask
+
+        Dask expect `.fast.evict()` to be availabe for manually triggering
```
Comment (pentschev) [inline dask_cuda/proxify_host_file.py:611]:
```suggestion
        Dask expects `.fast.evict()` to be availabe for manually triggering
```

**Review by pentschev — APPROVED** [review]
Just minor typo fixes and question not related to this PR itself, but otherwise LGTM. Thanks @madsbk ! 😄

File: dask_cuda/proxify_host_file.py (hunk 2 above)
Comment (madsbk) [reply dask_cuda/proxify_host_file.py:587]: I don't think that is necessary. Hopefully, we get to a point where Dask never calls this.

**Comment by pentschev** [comment]
@gpucibot merge

**Comment by pentschev** [comment]
Thanks @madsbk for fixing this!

**Pull request merged**